downloader/src/main/kotlin/Main.kt

```diff
@@ -198,58 +198,64 @@ object Main {
         p("Downloading source code to '${targetDir.absolutePath}'...")
 
         if (target.vcsProcessed.url.isNotBlank()) {
-            p("Trying to download from URL '${target.vcsProcessed.url}'...")
-
-            if (target.vcsProcessed.url != target.vcs.url) {
-                p("URL was normalized, original URL was '${target.vcs.url}'.")
-            }
+            return try {
+                downloadFromVcs(target, targetDir)
+            } catch (e: DownloadException) {
+                if (stacktrace) {
+                    e.printStackTrace()
+                }
 
-            if (target.vcsProcessed.revision.isBlank()) {
-                p("WARNING: No VCS revision provided, downloaded source code does likely not match revision " +
-                        target.version)
-            } else {
-                p("Downloading revision '${target.vcsProcessed.revision}'.")
+                p("Download from VCS failed. Falling back to source package ...")
+                // TODO: Clean up debris left by failed VCS download (i.e. a ".git" directory)
+                downloadSourcePackage(target, targetDir)
             }
+        } else {
+            p("No VCS URL provided.")
+            p("Trying to download source package ...")
+            return downloadSourcePackage(target, targetDir)
```

Comment:
Does it make sense (improve readability by simplifying code) to have only one call to `downloadSourcePackage()`, e.g. by moving the `return` into the `try` part, moving `downloadSourcePackage()` out of `catch` and after the `if` block, dropping the `else` branch?